src/test/java/org/broadinstitute/hellbender/utils/mcmc/GibbsSamplerSingleGaussianUnitTest.java

```diff
@@ -0,0 +1,181 @@
+package org.broadinstitute.hellbender.utils.mcmc;
+
+import com.google.common.primitives.Doubles;
+import org.apache.commons.math3.stat.descriptive.moment.Mean;
+import org.apache.commons.math3.stat.descriptive.moment.StandardDeviation;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Unit test for {@link GibbsSampler}.  Demonstrates application of {@link GibbsSampler} to a {@link ParameterizedModel}
+ * that is specified using {@link ParameterizedState} and {@link DataCollection}.
+ * <p>
+ *     Test performs Bayesian inference of a Gaussian model with 2 global parameters specifying the variance and the mean.
+ * </p>
+ * <p>
+ *     Data consists of a list of 10000 datapoints drawn from a normal distribution with unity variance and mean.
+ * </p>
+ * <p>
+ *     Success of the test is determined by recovery of the input variance and mean,
+ *     as well as agreement of the standard deviations of the parameter posteriors with those given by both the
+ *     python package emcee (see http://dan.iel.fm/emcee for details) and numerical evaluation in Mathematica
+ *     of the analytic forms of the posteriors.
+ * </p>
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public final class GibbsSamplerSingleGaussianUnitTest extends BaseTest {
+    private static final String TEST_SUB_DIR = publicTestDir + "org/broadinstitute/utils/mcmc/";
+
+    private static final File COVERAGES_FILE = new File(TEST_SUB_DIR
```

Comment:
Instead of hard-coding a test file, why not generate draws from a normal distribution on the fly?  The cost of sampling is probably less than I/O, so the test wouldn't take any longer.  In my opinion a bigger advantage is to make this test self-contained.